src/components/Insight/Comments.js

```diff
@@ -58,24 +63,44 @@ function editComment (id, content) {
   })
 }
 
-export default ({ id, authorId, count }) => {
+export default ({
+  id,
+  authorId,
+  count,
+  onEdit = editComment,
+  onDelete = deleteComment,
+  onCreate = createComment,
+  onGet = getComments
+}) => {
   const [comments, setComments] = useState([])
   const [loading, setLoading] = useState(true)
 
-  useEffect(() => {
-    getComments(id).then(({ data }) => {
+  const loadCommentsCount = () => {
+    onGet(id).then(({ data }) => {
       setComments(data.comments)
       setLoading(false)
     })
+  }
+
+  useEffect(() => {
+    loadCommentsCount()
   }, [])
 
+  const onClose = () => {
+    loadCommentsCount()
+  }
+
+  let showingCount = comments.length || count
```

Comment:
1. Value is never reassigned;
2. Better to call this variable `commentsCount`;
3. Should be initialised at the top of comment, otherwise it's lost between all hooks definition and `return` statement.